src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -516,6 +516,37 @@ readLimitField(unsigned int& limit, Tuning::LimitRange const& range,
     return boost::none;
 }
 
+boost::optional<Seed>
+parseRippleLibSeed(Json::Value const& params)
+{
+    // ripple-lib encodes seed used to generate an Ed25519 wallet in a
+    // non-standard way. While rippled never encode seeds that way, we
+    // try to detect such keys to avoid user confusion.
+    auto decode = [](Json::Value const& j) -> boost::optional<Seed>
+    {
+
+        if (!j.isString())
+            return boost::none;
+
+        auto const result = decodeBase58Token(j.asString(), TokenType::None);
```

Comment:
A passphrase isn't going to be base58 encoded. Shouldn't this use a different decode for passphrases?

Reply:
Several codepaths use “passphrase” as the catch-all (e.g. `wallet_propose`) and we make an effort to determine what this thing is.

I tried very carefully to keep the semantics sane and in line with previous expectations, but I’ll double-check.

Reply:
You're right, I'm looking at `ParseGenericSeed` (https://github.com/seelabs/rippled/blob/develop/src/ripple/protocol/impl/Seed.cpp#L95-L130), and `GetSeedFromRPC` (https://github.com/seelabs/rippled/blob/develop/src/ripple/rpc/impl/RPCHelpers.cpp#L542-L579) the logic is:

If any of the following are set: passphrase, seed, seed hex, try the following:
1) If "seed", parse as base58
2) If "passphrase":
   a) check if hex, otherwise
   b) check if base 58, otherwise
   c) use RFC1751, otherwise
   d) hash the passphrase
3) if "seed_hex", set hex exact

Reply:
Yeah, the logic is, let's be generous and say convoluted. I went through and reverified the logic and I'm convinced the changes do the right thing.

The only "potential" problem is someone that used a passphrase that _happens_ to also be a valid "ripple-lib encoded Ed25519" string. But the chances of that are, essentially, astronomical.